backend/data/src/main/kotlin/io/tolgee/formats/apple/in/xcstrings/XcstringsFileProcessor.kt

```diff
@@ -41,33 +44,77 @@ class XcstringsFileProcessor(
       context.addKeyDescription(key, comment)
     }
 
+    if (!localizations.has(sourceLanguage)) {
+      val converted = messageConvertor.convert(
+        rawData = key,
+        languageTag = sourceLanguage,
+        convertPlaceholders = context.importSettings.convertPlaceholdersToIcu,
+        isProjectIcuEnabled = context.projectIcuPlaceholdersEnabled
+      )
+
+      context.addTranslation(
+        keyName = key,
+        languageName = sourceLanguage,
+        value = converted.message,
+        convertedBy = importFormat,
+        state = TranslationState.TRANSLATED,
+        rawData = key,
+        pluralArgName = converted.pluralArgName
+      )
+    }
+
     localizations.fields().forEach { (languageTag, localization) ->
       when {
         localization.has("stringUnit") -> {
           processSingleTranslation(key, languageTag, localization)
         }
+
         localization.has("variations") -> {
           processPluralTranslation(key, languageTag, localization)
         }
       }
     }
   }
 
+
+  private fun convertXCtoTolgeeTranslationState(xcState: String?): TranslationState {
+    return when (xcState) {
+      "needs_review" -> TranslationState.TRANSLATED
+      "translated" -> TranslationState.REVIEWED
+      null -> TranslationState.UNTRANSLATED
```

Comment:
I don't think you need the null branch, when it goes to the same state in the else branch.